packages/transforms/rename/src/bareRename.ts

```diff
@@ -0,0 +1,76 @@
+import { GraphQLSchema, GraphQLFieldConfig } from 'graphql';
+import { MeshTransform, MeshTransformOptions, YamlConfig } from '@graphql-mesh/types';
+import { IResolvers, renameType, MapperKind, mapSchema } from '@graphql-tools/utils';
+import { mergeSchemas } from '@graphql-tools/merge';
+
+export default class RenameBareTransform implements MeshTransform {
+  noWrap = true;
+  typesMap: Map<string, string>;
+  fieldsMap: Map<string, { [name: string]: string }>;
+
+  constructor(options: MeshTransformOptions<YamlConfig.RenameTransform>) {
+    const { config } = options;
+    this.typesMap = new Map();
+    this.fieldsMap = new Map();
+
+    for (const rename of config.renames) {
+      const {
+        from: { type: fromTypeName, field: fromFieldName },
+        to: { type: toTypeName, field: toFieldName },
+      } = rename;
+      if (fromTypeName && !fromFieldName && toTypeName && !toFieldName && fromTypeName !== toTypeName) {
+        this.typesMap.set(fromTypeName, toTypeName);
+      }
+      if (fromTypeName && fromFieldName && toTypeName && toFieldName && fromFieldName !== toFieldName) {
+        const entry = { [fromFieldName]: toFieldName };
+        this.fieldsMap.set(
+          fromTypeName,
+          this.fieldsMap.has(fromTypeName) ? { ...this.fieldsMap.get(fromTypeName), ...entry } : entry
+        );
+      }
+    }
+  }
+
+  renameType(type: any) {
+    const newTypeName = this.typesMap.get(type.toString());
+    return newTypeName ? renameType(type, newTypeName) : undefined;
+  }
+
+  transformSchema(schema: GraphQLSchema) {
+    const renamedSchema = mapSchema(schema, {
+      [MapperKind.TYPE]: type => this.renameType(type),
+      [MapperKind.ROOT_OBJECT]: type => this.renameType(type),
+      [MapperKind.COMPOSITE_FIELD]: (
+        fieldConfig: GraphQLFieldConfig<any, any>,
+        fieldName: string,
+        typeName: string
+      ) => {
+        const newFieldName = this.fieldsMap.has(typeName) && this.fieldsMap.get(typeName)[fieldName];
+        return newFieldName ? [newFieldName, fieldConfig] : undefined;
+      },
+    });
+
+    // Root fields always have a default resolver and so don't need mapping
+    ['Query', 'Mutation', 'Subscription'].forEach(type => this.fieldsMap.delete(type));
+
+    const resolvers: IResolvers | IResolvers[] = [];
+
+    if (this.fieldsMap.size) {
+      for (const [type, fields] of this.fieldsMap.entries()) {
+        const typeName = this.typesMap.get(type) || type;
+        const typeResolvers = Object.entries(fields).reduce((fieldResolvers, [oldName, newName]) => {
+          fieldResolvers[newName] = (object: any) => object[oldName];
+          return fieldResolvers;
+        }, {});
+        resolvers.push({ [typeName]: typeResolvers });
+      }
+    }
+
+    return resolvers.length
+      ? mergeSchemas({
```

Comment:
You can do everything here within mapSchema by adjusting the returned fieldConfig as necessary.

Reply:
Just to keep this cleaner, up to you

Reply:
Thank you so much for the suggestion.
This is the first time I work with `graphql-tools` and from the documentation and example I looked at I didn't catch the opportunity to set resolvers in fieldConfig. Obviously, it makes perfect sense to do it like so.

This allowed for a significant simplification of implementation and code; so much cleaner!